clusterloader2/pkg/measurement/common/kcm_metrics.go

```diff
@@ -0,0 +1,223 @@
+				// since the leader will have larger workqueue backlog.
+				result := metricsFromHosts[0]
+				for i := 1; i < len(metricsFromHosts); i++ {
+					result = getMaxOrMin(result, metricsFromHosts[i], true)
+				}
+				// We keep tracking the smallest value that we have seen.
+				m.depthMetrics = getMaxOrMin(m.depthMetrics, result, false)
+			case <-m.stopCh:
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (m *kcmWorkQueueDepthMeasurement) createSummary() (measurement.Summary, error) {
+	perfData := &measurementutil.PerfData{
+		Version:   kcmWorkQueueDepthMetricVersion,
+		DataItems: []measurementutil.DataItem{},
+	}
+	for resource, value := range m.depthMetrics {
+		perfData.DataItems = append(perfData.DataItems, measurementutil.DataItem{
+			Data: map[string]float64{
+				"Min": value,
+			},
+			Labels: map[string]string{
+				"Resource": resource,
+			},
+		})
+	}
+	content, err := util.PrettyPrintJSON(perfData)
+	if err != nil {
+		return nil, err
+	}
+	if _, found := m.depthMetrics[hpaResource]; !found {
+		return nil, fmt.Errorf("can't find work queue depth metric for %v", hpaResource)
+	}
+	// If the min value is consistently larger than the allowed the workqueue backlog,
+	// it means the controller can't keep up with load for reconciliation.
+	if m.hpaAllowedBacklog != nil && m.depthMetrics[hpaResource] > *m.hpaAllowedBacklog {
+		err = errors.NewMetricViolationError(kcmWorkQueueDepthMetricName, fmt.Sprintf("HPA work queue backlog %v exceeds the allowed threshold %v", m.depthMetrics[hpaResource], *m.hpaAllowedBacklog))
+		klog.Errorf("%s: %v", m, err)
+	}
+	return measurement.CreateSummary(kcmWorkQueueDepthMetricName, "json", content), err
+}
+
+func (e *kcmWorkQueueDepthMeasurement) getKCMWorkQueueDepth(host string, provider provider.Provider, port int) (map[string]float64, error) {
+	cmd := fmt.Sprintf("curl -k https://localhost:%d/metrics", port)
+	samples, err := e.sshKCMMetrics(cmd, host, provider)
+	if err != nil {
+		return nil, err
+	}
+	workQueueDepth := map[string]float64{}
+	for _, sample := range samples {
+		if sample.Metric[model.MetricNameLabel] != "workqueue_depth" {
+			continue
+		}
+		resourceName, found := sample.Metric["name"]
+		if !found {
+			continue
+		}
+		workQueueDepth[string(resourceName)] = float64(sample.Value)
+	}
+	return workQueueDepth, nil
+}
+
+func (e *kcmWorkQueueDepthMeasurement) sshKCMMetrics(cmd, host string, provider provider.Provider) ([]*model.Sample, error) {
+	sshResult, err := measurementutil.SSH(cmd, host+":22", provider)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected error (code: %d) in ssh connection to master: %#v", sshResult.Code, err)
+	} else if sshResult.Code != 0 {
+		return nil, fmt.Errorf("failed running command: %s on the host: %s, result: %+v", cmd, host, sshResult)
+	}
+	data := sshResult.Stdout
+	return measurementutil.ExtractMetricSamples(data)
+}
+
+func getMaxOrMin(existing, new map[string]float64, max bool) map[string]float64 {
```

Comment:
You don't seem to use `max` param at all, this function always seems to return the `min`, do we want to strip the `max param` and change the function name to `getMin` ?